### ISSUE TITLE
Override coreness for arXiv

### ISSUE THREAD

**Issue by michamos** [inspirehep/inspirehep#1690]
When we harvest arXiv, we harvest some categories completely as CORE/non-core. This means the incoming records won't halt for selection in the holdingpen, so there's no opportunity for a curator to decide whether they are core. Instead, coreness is determined based on the category (for example `hep-*` is core, `nucl-*` is not). But for some of those record, the blanket assignment is incorrect, and we need to override it. The issue is that we don't want to delay ingestion, and instead do this step at a later stage. Ideally, this would look operationally like core assignement in the holdingpen, but would perform it on the already created record (and create curation tickets).

Additional info:
* core->non-core can be done manually so doesn't necessarily need to be handled, as it's low volume (<500 records/year).
* non-core -> core happens ~2.5k/year.

**Comment by michamos** [comment]
## Proposed solution

Right before completing the workflow for these records, we tag the auto-non-core records as `core selection` and we display them under the facet `Filter for Pending action` (very bottom), so they can easily be retrieved in the holdingpen.
We display `core`/`accept` buttons similar to what we do for records waiting for curator approval (minus the `reject` button, as the records are already in), and keep displaying all the metadata in the right-most holdingpen list column.
* Clicking on `accept` (=non-core) will fully complete the workflow and remove it from the set of workflows in `core selection`, and hide the buttons.
* Clicking on `core` will do the same, but will also start a new workflow of type `core_selection` (instead of `article`), which receives the `data` + `extra_data` payload of the current workflow and perfoms some of the same steps as in https://github.com/inspirehep/inspire-next/blob/f014987de7e252e0969d145f86a07bdac08a1831/inspirehep/modules/workflows/workflows/article.py#L548-L553 that have different effects for core records (e.g. keyword extraction, create curation ticket, etc.). It will do a request to `hep` to set `core: True` but won't do anything on legacy (the current approach can be used as long as legacy is running).

@ksachs could you comment on this proposal?

**Comment by ksachs** [comment]
To make sure i understand this:
"right before completing the workflow" means after ingestion/send to legacy.
After clicking `core` the rest is automatic to get the necessary information into the record.

To me that makes sense @fschwenn ?

**Comment by ksachs** [comment]
keep in mind we will have a similar case for journals we harvest automatically.
It would be cool to get those in directly and add `core` and subject later.

**Comment by ksachs** [comment]
@fschwenn and I would at some point like to see arXiv records cross-listed to hep*, i.e. records that are automatically tagged as `core` but are not primarc hep*. Sometimes we want to change subject and `core` which can be done via the editor.
Something to keep them in the holdingpen would be cool, otherwise we set up a search to look for them.

**Comment by michamos** [comment]
> "right before completing the workflow" means after ingestion/send to legacy.
After clicking core the rest is automatic to get the necessary information into the record.

Exactly. However, our idea was that you would use this interface only after switching off legacy, as you have a different approach that works on legacy for now. Would you need it earlier?

> keep in mind we will have a similar case for journals we harvest automatically.
It would be cool to get those in directly and add core and subject later.

For those we grey-out the `reject` button, as it's less time-sensitive. We could modify it in the future to use the same mechanism as for arXiv non-core, but is it ok if we leave that for later?

> @fschwenn and I would at some point like to see arXiv records cross-listed to hep*, i.e. records that are automatically tagged as core but are not primarc hep*. Sometimes we want to change subject and core which can be done via the editor.
Something to keep them in the holdingpen would be cool, otherwise we set up a search to look for them.

We can flag them in the holdingpen, but if you need to edit them manually anyway it looks more cumbersome to me than to do a search in INSPIRE.

**Comment by ksachs** [comment]
> Exactly. However, our idea was that you would use this interface only after switching off legacy, as you have a different approach that works on legacy for now. Would you need it earlier?

that's fine. Might be good to test it before we switch off lecacy :-)

> For those we grey-out the reject button, as it's less time-sensitive. We could modify it in the future to use the same mechanism as for arXiv non-core, but is it ok if we leave that for later?

That's fine

> We can flag them in the holdingpen, but if you need to edit them manually anyway it looks more cumbersome to me than to do a search in INSPIRE

I assume book-keeping is easier in the holdingpen, so it's clear Florian or I saw the record but only once.

**Comment by fschwenn** [comment]
The proposed solution looks very good to me. I fully agree with Kirsten in all points.

**Comment by drjova** [comment]
We create a new workflow `hep_core` which does the following steps:
1. fetch latest record from hep
2. change the flag to core
3. add keywords 
4. creates a RT ticket
5. complete

On the current `article` workflow, we complete the workflow normally and we start `hep_core` workflow as `HALTED` and we copy all the data. The new workflow should be "Pending for Action" and when the curator clicks "CORE" we continue the `hep_core` workflow with the steps described above.

**Comment by drjova** [comment]
* Create a new workflows https://github.com/inspirehep/inspire-next/tree/master/inspirehep/modules/workflows/workflows
 1. Create a step to retrieve latest record from hep and save it into the workflow object
 2. Add core step `add_core`  
 3. Run `classify_papper` https://github.com/inspirehep/inspire-next/blob/f3a85e38c87dea44c41f7debdd9553a13c2e993e/inspirehep/modules/workflows/workflows/article.py#L188
 4. `prepare_keywords` https://github.com/inspirehep/inspire-next/blob/f3a85e38c87dea44c41f7debdd9553a13c2e993e/inspirehep/modules/workflows/tasks/submission.py#L332
 5.  Create ticket as https://github.com/inspirehep/inspire-next/blob/f3a85e38c87dea44c41f7debdd9553a13c2e993e/inspirehep/modules/workflows/workflows/article.py#L266-L273
 6. Add store record https://github.com/inspirehep/inspire-next/blob/f3a85e38c87dea44c41f7debdd9553a13c2e993e/inspirehep/modules/workflows/workflows/article.py#L387
